app/scripts/background.js

```diff
@@ -721,10 +724,28 @@ function setupController(initState, initLangCode) {
         ),
       );
 
-    // Finally, reject all approvals managed by the ApprovalController
+    // Finally, resolve snap dialog approvals on Flask and reject all the others managed by the ApprovalController.
+
+    ///: BEGIN:ONLY_INCLUDE_IN(flask)
+    Object.values(controller.approvalController.state.pendingApprovals).forEach(
+      ({ id, type }) => {
+        if (type.startsWith(RestrictedMethods.snap_dialog)) {
+          controller.approvalController.accept(id, null);
+        } else {
+          controller.appStateController.reject(
```

Comment:
Is this needed if we have the below `clear()` in Flask too? Maybe we don't need to fence the `clear` then 🤔